sql/core/src/test/scala/org/apache/spark/sql/FileBasedDataSourceSuite.scala

```diff
@@ -470,22 +471,24 @@ class FileBasedDataSourceSuite extends QueryTest with SharedSQLContext with Befo
   }
 
   test("SPARK-25237 compute correct input metrics in FileScanRDD") {
-    withTempPath { p =>
-      val path = p.getAbsolutePath
-      spark.range(1000).repartition(1).write.csv(path)
-      val bytesReads = new mutable.ArrayBuffer[Long]()
-      val bytesReadListener = new SparkListener() {
-        override def onTaskEnd(taskEnd: SparkListenerTaskEnd) {
-          bytesReads += taskEnd.taskMetrics.inputMetrics.bytesRead
+    withSQLConf(SQLConf.USE_V1_SOURCE_READER_LIST.key -> "csv") {
+      withTempPath { p =>
+        val path = p.getAbsolutePath
+        spark.range(1000).repartition(1).write.csv(path)
+        val bytesReads = new mutable.ArrayBuffer[Long]()
+        val bytesReadListener = new SparkListener() {
+          override def onTaskEnd(taskEnd: SparkListenerTaskEnd) {
+            bytesReads += taskEnd.taskMetrics.inputMetrics.bytesRead
+          }
+        }
+        sparkContext.addSparkListener(bytesReadListener)
+        try {
+          spark.read.csv(path).limit(1).collect()
+          sparkContext.listenerBus.waitUntilEmpty(1000L)
+          assert(bytesReads.sum === 7860)
+        } finally {
+          sparkContext.removeSparkListener(bytesReadListener)
         }
-      }
-      sparkContext.addSparkListener(bytesReadListener)
-      try {
-        spark.read.csv(path).limit(1).collect()
-        sparkContext.listenerBus.waitUntilEmpty(1000L)
-        assert(bytesReads.sum === 7860)
-      } finally {
-        sparkContext.removeSparkListener(bytesReadListener)
       }
     }
   }
```

Comment:
If this fails due to the number of bytes, can we have test coverage for both? Note that only `try` part is changed.
```scala
  test("SPARK-25237 compute correct input metrics in FileScanRDD") {
    withTempPath { p =>
      val path = p.getAbsolutePath
      spark.range(1000).repartition(1).write.csv(path)
      val bytesReads = new mutable.ArrayBuffer[Long]()
      val bytesReadListener = new SparkListener() {
        override def onTaskEnd(taskEnd: SparkListenerTaskEnd) {
          bytesReads += taskEnd.taskMetrics.inputMetrics.bytesRead
        }
      }
      sparkContext.addSparkListener(bytesReadListener)
      try {
        Seq(("csv", 7860), ("", 3930)).foreach { case (list, sum) =>
          bytesReads.clear()
          withSQLConf(SQLConf.USE_V1_SOURCE_READER_LIST.key -> list) {
            spark.read.csv(path).limit(1).collect()
            sparkContext.listenerBus.waitUntilEmpty(1000L)
            assert(bytesReads.sum === sum)
          }
        }
      } finally {
        sparkContext.removeSparkListener(bytesReadListener)
      }
    }
  }
```

Reply:
See my comment in https://github.com/apache/spark/pull/24005#discussion_r268051553

Reply:
So, we had better check both cases.

Reply:
Sorry but I don't think testing the V2 now is necessary. I will add a TODO here.